### ISSUE TITLE
[Package Request] UVNC repeater

### ISSUE THREAD

**Issue by andersenco** [SynoCommunity/spksrc#2999]
Name: UVNC repeater
Description: UltraVNC repeater implementation on Synology
Website: http://www.uvnc.com/products/uvnc-repeater.html
Software documentation: 
Build/Installation documentation: https://github.com/tenchman/uvncrepeater-ac
Source Code: https://github.com/tenchman/uvncrepeater-ac
License: Unsertain
Original code: https://web.archive.org/web/20160301191943/http://jtko.mbnet.fi/ (website no longer available)

With some help I would problably be able to do some of it by myself.
Using https://github.com/SynoCommunity/spksrc/wiki/Developers-HOW-TO how do I define download from Github in the MakeFile?

**Comment by andersenco** [comment]
I have started out and got to the Makefile in the spksrc/cross directory.

I want to be able to run 'make' in work_dir but it seems to be running in a loop.
Makefile:
`PKG_NAME = repeater
PKG_VERS = 014
PKG_EXT = zip
PKG_DIST_NAME = $(PKG_NAME)$(PKG_VERS).$(PKG_EXT)
PKG_DIST_SITE = http://web.archive.org/web/20150427093843/http://koti.mbnet.fi/jtko/uvncrepeater/
PKG_DIR = Ver$(PKG_VERS)

DEPENDS = 

HOMEPAGE = http://koti.mbnet.fi/jtko/uvncrepeater/
COMMENT  = Ported ultraVNC repeater for Linux
LICENSE  = 

GNU_CONFIGURE = 0
CONFIGURE_ARGS = 

CONFIGURE_TARGET = nop
COMPILE_TARGET = myComp

include ../../mk/spksrc.cross-cc.mk

myComp:
	$(MSG) "Building Repeater for host system"
	cd $(WORK_DIR)/Ver$(PKG_VERS) 
	make
`

**Comment by chickenandpork** [comment]
You want to run it with a target matching your architecture. For example:

    make arch-avoton-6.1

Check the build docs again.  I don’t know why it would be looping on you but you should build as: (I think):

    cd spksrc
    make setup  (make local.mk ?)
    cd cross/repeater
    make arch-avoton-6.1

It would be awesome to just “make” in your workdir but I don’t see that working.

**Comment by andersenco** [comment]
Hi
The sourcecode makefile is different from Makefiles for the actual packages.
I'm still learning the syntax for these and found out I had to put the make sttement on the same line as my cd $(...
(I did not use .phony)

So I got this part working late last night and now I can start on the actual package ;-)

Thank you for your respons.

Best regards, Morten

**Comment by andersenco** [comment]
Hi again
I have just tried compiling the spk part.
I have stripped the new Makefile for more or less everything for now, and get some errors (Makefile below). As an example I have used Transmission which has similar errors:

`
cat: /spksrc/spk/repeater/work-armada370-6.1/python-cc.mk: No such file or directory
===>  Creating target installation dir of repeater
(cd /spksrc/spk/repeater/work-armada370-6.1/install//var/packages/repeater/target && tar cpf - `cat /spksrc/spk/repeater/work-armada370-6.1/PLIST | cut -d':' -f2`) | \
  tar xpf - -C /spksrc/spk/repeater/work-armada370-6.1/staging
/bin/sh: 1: cd: can't cd to /spksrc/spk/repeater/work-armada370-6.1/install//var/packages/repeater/target
tar: This does not look like a tar archive
tar: Exiting with failure status due to previous errors
../../mk/spksrc.copy.mk:46: recipe for target 'copy_target' failed
make[1]: *** [copy_target] Error 2
make[1]: Leaving directory '/spksrc/spk/repeater'
../../mk/spksrc.spk.mk:452: recipe for target 'arch-armada370' failed
make: [arch-armada370] Error 2 (ignored)
`

Makefile:
`
SPK_NAME = repeater
SPK_VERS = 014
SPK_REV = 1

DEPENDS = 

MAINTAINER = Andersenco
DESCRIPTION = UltraVNC repeater for Synology
RELOAD_UI = yes
STARTABLE = yes
DISPLAY_NAME = Repeater
CHANGELOG = "1. Use DSM 5+6 generic service approach for correct permissions."

HOMEPAGE = 
LICENSE  = 

include ../../mk/spksrc.spk.mk
`

**Comment by chickenandpork** [comment]
TL;DR: maybe `DEPENDS = cross/$(SPK_NAME)`

### Markup ###
I would suggest you actually make a PR, mark it as "Work In Progress", and keep it updated* to help show your diff in an easy-to-read format that we're used to.  As all new code, it would be green, but easy to read.  People like @ymartin59 and @Diaoul have near-zero spare time, make it efficient.  Of course, the bigger contributors might have better suggestions, this is what works for me outside of `spksrc`

### "keep it updated" ###
personally, I tend to do unpunished work as an initial checkout with later fixups.  After the first `git commit -am "message"`, I add fixes with `git commit -a --fixup HEAD && git rebase -I --autosquash HEAD^^ && git push --force origin`. My PR tends to be from my fork as "origin" to the upstream spksrc.

### Repeater? ###

"repeater" is fairly generic; wouldn't "uvncrepeater" make a more precise name?  Would people wanting it search for "UVNC Repeater", or would they search for "VNC" and find "VNC Repeater"?  I guess I'm trying to suggest what can help your work be found and leveraged most widely once your hard work is merged and pushed.

### Your current build issue ###
```
DEPENDS =
```

This is concerning; you may need to have (I'm unsure whether it's implicit):

```
DEPENDS = cross/$(SPK_NAME)
```
What you're seeing is that the build doesn't change to a directory because the directory probably doesn't exist.  I believe it's created when the first DEPEND item is installed after build.  Because your build process doesn't have a dependency, nothing is built, so nothing installs, so nothing creates the install directory it's trying to change to.  Now, I see examples like `couchpotatoserver` that doesn't have it, but a quick search for `fgrep 'cross/$(SPK' spk/*/Makefile` shows many which do, so I'm not 100% sure.  Try it?
